packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -413,6 +413,27 @@ moduleFor('Components test: curly components', class extends RenderingTest {
     assert.equal($span.attr('class'), 'inner');
   }
 
+  ['@test a void element does not have childNodes'](assert) {
+    let fooBarInstance;
+    let FooBarComponent = Component.extend({
+      tagName: 'input',
+      init() {
+        this._super();
+        fooBarInstance = this;
+      }
+    });
+
+    this.registerComponent('foo-bar', { ComponentClass: FooBarComponent, template: '{{yield}}' });
+
+    this.render('{{foo-bar}}');
+
+    assert.deepEqual(fooBarInstance.element.childNodes.length, 0);
+
+    this.runTask(() => this.rerender());
+
+    assert.deepEqual(fooBarInstance.element.childNodes.length, 0);
```

Comment:
When comparing against 0, it's always a good bet to use `assert.strictEqual` which uses === instead of == :)